tests/test_enterprise/api/test_views.py

```diff
@@ -627,20 +614,175 @@ def test_enterprise_customer_courses(
 
         assert response == expected
 
+    @ddt.data(
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            False, False,
+            '',
+            {
+                'detail': "User must be a staff user or "
+                          "associated with the endpoint's appointed common Enterprise Customer."
+            },
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            False, True,
+            '',
+            {
+                'detail': "User must be a staff user or "
+                          "associated with the endpoint's appointed common Enterprise Customer."
+            },
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            True, False,  # Is a staff user, so it's allowed.
```

Comment:
Ok from this I realized I have to change the test logic slightly: instead of `has_linked_user`, it should be `is_staff`, and the logic that creates the user should be inside of the `has_existing_catalog` branch.